File: Algoritmos/insertion_sort.cpp
#include <iostream>
using namespace std;

int* Insertion_sort(int array[], int size){
    int key; //Esta es la variable que se busca acamodar en si sitio en cada interación.
    int j;
    for(int i = 1; i<size;i++ ){
    key = array[i];
    j=i-1; //Esta nos sirve para hacer las comparaciónes entre el key y el elemento que esta a la izquiera. 
    while(key < array[j] && j >=0){ //Este ciclo comprueba que el elemento a la izquierda  
        array[j+1]=array[j];
        j=j-1;
    }
    array[j+1]=key;
    }
    return array;
}
int main(){
    int array[] = {10,10,10,10,10,10,10,10,10,10,10,0};
    int size = sizeof(array)/sizeof(array[0]); 
    int* ptr = Insertion_sort(array, size);
    for(int i=0; i<size ;i++){
        cout<<*(ptr+i)<<" ";
    }
}

File: Algoritmos/insertion_sort.py
def Insertion_sort(array):
    for i in range(1,len(array)):
        key = array[i]
        j = i-1
        while(key < array[j] and j >=0):
            array[j+1]=array[j]
            j=j-1
        array[j+1]=key
    return array

array = [4,4,0,12,1,1,1,1,1,40,1000,40,22,42,51,1]
sorted_array = Insertion_sort(array)
for i in range(0,len(array)):
    print(sorted_array[i]," ")
    



